lib/ansible/galaxy/oauth_device_flow.py

```diff
@@ -0,0 +1,124 @@
+# (c) 2020 Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import json
+import textwrap
+import time
+
+from ansible import context
+from ansible.errors import AnsibleError
+from ansible.galaxy.user_agent import user_agent
+from ansible.module_utils.six.moves.urllib.error import HTTPError
+from ansible.module_utils.urls import open_url
+from ansible.utils.color import stringc
+from ansible.utils.display import Display
+from collections import Mapping
+
+display = Display()
+
+
+class GitHubOAuthTokenCreator(object):
+    '''
+    Uses OAuth 2.0 device auth grant (RFC8628) to create limited user authorization token in an
+    out-of-band browser session using the GitHub OAuth device flow as described in
+    https://docs.github.com/en/developers/apps/authorizing-oauth-apps#device-flow
+    '''
+    GITHUB_DEVICE_AUTH_URL = 'https://github.com/login/device/code'
+    GITHUB_TOKEN_POLL_URL = 'https://github.com/login/oauth/access_token'
+    GITHUB_DEVICE_AUTH_HEADERS = {
+        "Accept": "application/vnd.github.v3+json,application/vnd.github.machine-man-preview+json,text/json,application/json",
```

Comment:
`machine-man` is not related to the device auth flow (it used to be necessary for GitHub Apps).

Besides, they've made those APIs stable already: https://developer.github.com/changes/2020-08-20-graduate-machine-man-and-sailor-v-previews/.